src/dotnet/HQ.Server/Services/ChargeCodeServiceV1.cs

```diff
@@ -102,4 +102,69 @@ public async Task<Result<GetChargeCodesV1.Response>> GetChargeCodesV1(GetChargeC
 
         return response;
     }
+
+    public async Task<string> GenerateNewChargeCode(WorkType workType, CancellationToken ct = default)
+    {
+       
+        string prefix = workType switch
+        {
+            WorkType.Project => "P",
+            WorkType.Quote => "Q",
+            WorkType.Service => "S",
+            _ => throw new ArgumentOutOfRangeException(nameof(workType), workType, null)
+        };
+        try
+        {
+            var latestChargeCode = await _context.ChargeCodes
+            .Where(c => c.Code.StartsWith(prefix))
+                    .OrderByDescending(c => c.Code)
+                    .FirstOrDefaultAsync(ct);
+            string nextChargeCode = "";
+            if (latestChargeCode == null)
+            {
+                switch (workType)
+                {
+                    case WorkType.Project:
+                        nextChargeCode = "P0000";
+                        break;
+                    case WorkType.Quote:
+                        nextChargeCode = "Q0000";
+                        break;
+                    case WorkType.Service:
+                        nextChargeCode = "S0000";
+                        break;
+
+                }
+            }
```

Comment:
WorkType enum can go away, we already have an enum for that on the ChargeCode entity - `ChargeCodeActivity`. 

Instead of doing all this string splitting, based it off the type being created and its associated number. It's less that we're incrementing the charge code itself but more the _related entity_.

Ex. if we are creating a project, we should set its project number to `MAX(ProjectNumber) + 1`, _then_ once we have that project number we can simply generate a `P` charge code record with the project number suffixed after it. The ProjectNumber has a unique constraint and we should be atomically creating the project record to ensure uniqueness.